lib/fluent/supervisor.rb

```diff
@@ -190,6 +190,12 @@ def install_supervisor_signal_handlers
         $log.debug 'fluentd supervisor process got SIGUSR2'
         supervisor_sigusr2_handler
       end
+
+      term_handler = trap :TERM do
+        # After SIGTERM, disable the sigdump file by ignoring SIGCONT.
+        trap(:CONT, nil)
+        term_handler.call
```

Comment:
Actually, `term_handler` may be a string.
I will add a check to see if it is a `Proc`.

Reply:
I see! This can be `String` or `Proc` or `nil according to the following document.

https://docs.ruby-lang.org/ja/latest/method/Signal/m/trap.html

> I will add a check to see if it is a Proc.

So this looks good!